contrib/sparse-checkout/update-google-cloud-cpp.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+echo "Using sparse checkout for google-cloud-cpp"
+
+FILES_TO_CHECKOUT=$(git rev-parse --git-dir)/info/sparse-checkout
+echo '!/*' > $FILES_TO_CHECKOUT
+echo '/google/*' >> $FILES_TO_CHECKOUT
```

Comment:
The idea is to checkout only the minimal set of files needed to build google-cloud-api. `google/` contains all libraries so we would not gain a lot. Maybe you could try to restrict this further?

Reply:
I've done my best at [`8d38047` (#72205)](https://github.com/ClickHouse/ClickHouse/pull/72205/commits/8d380470e098f4efc6fef1a59218077e3e872419)

The google-cloud-cpp dir goes from 5.2GB to 12M with sparse checkout, so not bad. But I'm afraid that the moment we add something else apart of KMS, it'll fail with the sparse checkout. Do we test this flavor somehow on the CI?

Reply:
A reduction to 12M is totally fine.

To test, you could
- `git clone` the repository (but don't check out submodules)
- switch the local code state to this PR
- then run `./contrib/update-submodules.sh` to do a sparse-checkout of the submodules
- then compile and check if something breaks.

Reply:
Yep, I did test locally before pushing changes. I meant if we test this regularly on the CI or something. Because the moment we add anything else apart from KMS, unless we keep updating the sparse checkout for this, or it'll end up broken